### ISSUE TITLE
Aggregates in select list result in model field type interface{}

### ISSUE THREAD

**Issue by myles-mcdonnell** [kyleconroy/sqlc#466]
db: Postgres

consider:

```    
select
   sum(it.vatpence) as vatpence
from ...
```

where `vatpence` is an integer SQLC generates a row model with `interface{}` as the field type.  Workaround is to cast to int64 in client code.

thx
M

**Comment by Maxiride** [comment]
> where `vatpence` is an integer SQLC generates a row model with `interface{}` as the field type. Workaround is to cast to int64 in client code.

Did you modify the generated sqlc code substituting the interface with an int or did you manually cast the interface to an int?

At the best of my knowledge I did it this way where `intCast `is the receiving end of the new int and `interfaceVar `is the interface returned by the sqlc query.:

```go
	intCast, ok := interfaceVar.(int)
	if !ok {
            ....
	}
```

Any feedback on this approach? I'm not sure its the best, nor the only way to do it.

**Comment by powersjcb** [comment]
@kyleconroy Do we have a list of query types that are not fully supported? It might be nice to have a note in the readme and examples of handling some of the current edge cases.

**Comment by kyleconroy** [comment]
We don't have such a list. There's a list of supported functions in [pg_catalog.go](https://github.com/kyleconroy/sqlc/blob/master/internal/postgresql/pg_catalog.go), but's it's not complete. We'll need to generate that list from a fresh install of PostgreSQL.

**Comment by kyleconroy** [comment]
This now works! I'm not sure when it got fixed, but v1.9.0 will generate the correct code, as [shown here](https://play.sqlc.dev/p/dd7e5b294c3c36bb2ca9263cef60c1c6249cd3b87e288e4040e0c3bf9cabdbd0).
